tap_salesforce/client.py

```diff
@@ -194,7 +196,7 @@ def get_records(
             yield from self._paginate(
                 "GET", f"/services/data/{self._API_VERSION}/queryAll/", params={"q": query}
             )
-        except requests.exceptions.HTTPError as e:
+        except QueryTimeoutError as e:
```

Comment:
There is no code that raises this error, meaning that this code will never be executed. I think you misunderstood what I meant in our earlier conversation.

Using `requests.exceptions.HTTPError` without checking _which_ error actually resulted in it, means that this code will essentially run everytime _anything_ error related happens in our http request. This is not what we want to do.

Instead, you need to look at the payload of the specific response and ensure that you are _only_ processing the situation where the result is due to a query timeout, not a normal http timeout.

the code should look something like this:

```python
except requests.exceptions.HTTPError as e:
    print(e.response.text)
```

I recommend running it again, putting a breakpoint on the error and then once you have the object, you know what to use.

Reply:
Ahh, my bad. I see the raise below. This should most likely still be handled differently. Gimme one second.

Reply:
The thing is that `backoff` decorator is hiding away the error. The payload doesn't raise to the caller. That is why I needed to raise the error myself